File: business_logic/src/test/java/org/metplus/curriculum/domain/job/JobRepositoryTest.java
package org.metplus.curriculum.domain.job;

import org.assertj.core.api.Assertions;
import org.junit.Before;
import org.junit.Test;
import org.metplus.curriculum.domain.DomainFactories;
import org.metplus.curriculum.domain.Job;

public abstract class JobRepositoryTest {

    private JobRepository repo;

    protected abstract JobRepository getRepository();

    @Before
    public void setup() {
        repo = getRepository();
    }

    @Test
    public void findById_it_retrieves_the_job_when_job_exists() {
        Job job = repo.save(DomainFactories.buildJob());
        Job savedJob = repo.findById(job.getJobId());
        Assertions.assertThat(savedJob).isEqualToComparingFieldByField(job);
    }
}

File: business_logic/src/test/kotlin/org/metplus/curriculum/domain/job/JobRepositoryTest.kt
package org.metplus.curriculum.domain.job

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.metplus.curriculum.domain.DomainFactories.buildJob

abstract class JobRepositoryTest {

    private lateinit var repo: JobRepository

    abstract val repository: JobRepository

    @Before
    fun setup() {
        repo = repository
    }

    @Test
    fun `findById it retrieves the job when job exists`() {
        val job = repo.save(buildJob(
            jobId = "some id"
        ))
        val savedJob = repo.findById(job.jobId)
        assertThat(savedJob).isEqualToIgnoringGivenFields(
            job,
            "jobId"
        )
        assertThat(savedJob!!.jobId).isNotEmpty()
    }

    @Test
    fun `save with existing jobId, it updates the job`() {
        repo.save(buildJob(jobId = "some job id"))
        val savedJob = repo.save(buildJob(
            jobId = "some job id",
            title = "some other title",
            description = "some other title"
        ))

        assertThat(savedJob).isEqualToComparingFieldByField(buildJob(
            jobId = "some job id",
            title = "some other title",
            description = "some other title"
        ))
    }
}